include/internal/csv_row.cpp

```diff
@@ -116,7 +116,7 @@ namespace csv {
         return this->buffer->split_buffer[this->start + n];
     }
 
-    HEDLEY_NON_NULL(1)
+    HEDLEY_NON_NULL(2)
```

Comment:
Argument `1` for the `__nonnull__` attribute refers to the implicit `this` parameter.

Reply:
Thanks for pointing this out. I never realized g++ added `this` as an artificial parameter.

For future reference: https://gcc.gnu.org/ml/gcc/2006-04/msg00549.html